colossalai_platform/cli/commands/projects/train.py

```diff
@@ -1,38 +1,74 @@
 import os
-from patch import patch_tensorboard
+import argparse
+import types
+from patch import patch_platform_specific_dependencies
 
-# ==================================
-# Code generated by ColossalAI-Platform-CLI. DO NOT EDIT.
+# Please do not remove this call,
+# the platform's runtime environment needs it.
+patch_platform_specific_dependencies()
+
+# ===================================================================
+# Example code generated by ColossalAI-Platform-CLI.
 #
-# ColossalAI Platform Specification
-# Start
-# ==================================
+# Modify it to fit your needs.
+# ===================================================================
 
-# patch dependencies
-patch_tensorboard()
 
-# Environment variables for your data access
-PROJECT_DIR = os.environ['PROJECT_DIR']
-DATASET_DIR = os.environ['DATASET_DIR']
-OUTPUT_DIR = os.environ['OUTPUT_DIR']
-INPUT_MODEL_DIR = os.environ['MODEL_DIR']
-CHECKPOINT_DIR = os.path.join(OUTPUT_DIR, 'checkpoint')
-TENSORBOARD_DIR = os.path.join(OUTPUT_DIR, 'tensorboard')
+def parse_args(parser: argparse.ArgumentParser):
+    # required arguments
+    parser.add_argument(
+        "--epoch",
+        type=int,
+        required=True,
+        help="The training epoch.",
+    )
+    parser.add_argument(
+        "--project_dir",
+        type=str,
+        required=True,
+        help="The directory contains the project code.",
+    )
+    parser.add_argument(
+        "--dataset_dir",
+        type=str,
+        required=True,
+        help="The directory contains the training dataset.",
+    )
+    parser.add_argument(
+        "--output_dir",
+        type=str,
+        required=True,
+        help="The directory project would write output into.",
+    )
 
+    # optional arguments, add more if you need
+    parser.add_argument(
+        "--model_dir",
+        type=str,
+        default=None,
+        help="The directory contains the model to finetune.",
+    )
+    return parser.parse_args()
```

Comment:
Do not run parse_args for the user, sometimes, they may want to do it on their own. Only adding arguments to the parser will be sufficient.

Reply:
Good.

Reply:
Updated.